packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1146,17 +1147,30 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
         if (newChild != null) {
           _childElements[index] = newChild;
           final SliverMultiBoxAdaptorParentData parentData = newChild.renderObject.parentData;
+          if (index == 0) {
+            parentData.layoutOffset = 0.0;
```

Comment:
In the case of underflow, it is possible that the child list only has one child with null layoutoffset. It can only happen when added an element into an empty sliver list. In this case, the child should be in index 0.